### PR TITLE
Feat/5 mirroring accounts method

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
# Pull Request: Mirroring Accounts Functionality

## Description

This pull request introduces the `mirror_account_chart` method, which synchronizes remote CashCtrl accounts with a desired target state provided as a DataFrame. The functionality includes options to delete accounts on the remote that are not present in the target DataFrame and to update modified accounts.

### Changes

- Added the `mirror_account_chart` method to handle the synchronization of accounts.
- Included a fixture `add_and_delete_vat_code` to create a VAT code at the start of the test and delete it at the end.
- Implemented the `test_mirror_accounts` function to test the mirroring functionality with various scenarios.
Here's the GitHub pull request markdown based on the changes provided:

### ChatGPT review:
<details> <summary>Click for ChatGPT prompt</summary>
Comparing the provided code with Python community standards, here are some key observations and feedback:

1. **Docstrings and Comments**:
   - **Positive**: The code includes docstrings for the `mirror_account_chart` method, which is good practice for explaining the purpose and usage of the function.
   - **Improvement**: Adding docstrings to the test functions and fixture would improve readability and maintainability.

2. **PEP 8 Compliance**:
   - **Positive**: The code generally follows PEP 8 conventions, including spacing, indentation, and line length.
   - **Improvement**: The comments within the code can be more descriptive, especially before complex logic blocks.

3. **Code Structure and Modularity**:
   - **Positive**: The code is modular, with clear separation between the core functionality (`mirror_account_chart`) and the testing code.
   - **Improvement**: Consider breaking down larger functions into smaller, more manageable pieces. For example, the `mirror_account_chart` method could be split into smaller methods for deleting, adding, and updating accounts.

4. **Use of Pandas**:
   - **Positive**: The use of Pandas for DataFrame operations is efficient and aligns with common practices in data handling.
   - **Improvement**: Ensure that the Pandas operations are optimized for performance, especially if dealing with large datasets.

5. **Error Handling**:
   - **Improvement**: The code currently lacks error handling. Adding try-except blocks to manage potential errors (e.g., issues with DataFrame operations or remote API calls) would make the code more robust.

6. **Testing**:
   - **Positive**: The use of pytest for fixture and test functions is a good practice and aligns with community standards for testing.
   - **Improvement**: Consider adding more test cases to cover edge scenarios and potential failure points. Also, include assertions to validate the state before and after significant operations.

7. **Code Readability**:
   - **Positive**: The code is generally readable with clear variable names and logical flow.
   - **Improvement**: Adding more inline comments to explain complex logic can enhance readability for other developers.

### Short Feedback Summary

The provided code largely adheres to Python community standards, with proper use of docstrings, modularity, and adherence to PEP 8 guidelines. To further align with best practices, consider adding more descriptive comments, improving error handling, breaking down large functions into smaller pieces, and expanding test coverage. This will enhance readability, maintainability, and robustness of the code.

</details>

@lasuk  Please review the changes and provide any feedback. Thank you!

### CODE REVIEW

File: tests/test_account_chart.py [hunk 1]
```diff
@@ -185,3 +205,50 @@ def test_update_account_with_invalid_group_raise_error():
         cashctrl_ledger.update_account(account=1149, currency='USD',
             text='test create account', vat_code='MwSt. 2.6%', group='/ABC'
         )
+
+# Tests the mirroring functionality of accounts.
+def test_mirror_accounts(add_and_delete_vat_code):
+    target_df = (pd.read_csv('tests/initial_accounts.csv', skipinitialspace=True))
```
Comment (lasuk) [inline tests/test_account_chart.py:211]: The CashCtrl account chart is pretty hard to read in csv format because it is defined in multiple languages.

How can I know it is different from the default chart on the server?
(Otherwise, the test would only check whether the data remains unchanged)

As an alternative, we could use `TestLedger.account_chart()` defined on [pyledger/test_ledger.py#L17-L80](
https://github.com/macxred/pyledger/blob/7d28cbea70c5b99b527fe6db296473563beffa75/pyledger/test_ledger.py#L17-L80). We would need to add a 'group' column to this account chart, tough.

Let's discuss this proposal in tomorrow's call.

File: tests/test_account_chart.py [hunk 2]
```diff
@@ -185,3 +205,50 @@ def test_update_account_with_invalid_group_raise_error():
...
+    # Mirror target accounts onto server with delete=True
+    cashctrl_ledger.mirror_account_chart(target_df, delete=True)
+    mirrored_df = cashctrl_ledger.account_chart().reset_index()
+    m = standardized_df.merge(mirrored_df, how='outer', indicator=True)
+    assert (m['_merge'] == 'both').all(), (
+            'Mirroring error: Some target accounts were not mirrored'
+        )
+
+    # Updating account that has VAT code to avoid error
+    target_df.loc[target_df.index[0], 'text'] = "New_Test_Text"
```
Comment (lasuk) [inline tests/test_account_chart.py:235]: I am not following this code block. 

What does this expression do?
```
target_df.loc[target_df.index[0], ...]
```

Maybe also clarify comment.

**Review by lasuk — COMMENTED** [review]
Thanks. Looks very good. 

I left a few questions/comments.

File: tests/test_account_chart.py (hunk 2 above)
Comment (lasuk) [reply tests/test_account_chart.py]: Thanks for the clarification in 3d9fcf6
